src/main/java/org/springframework/cloud/deployer/scheduler/spi/kubernetes/ContainerCreator.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.scheduler.spi.kubernetes;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.EnvVar;
+import io.fabric8.kubernetes.api.model.batch.CronJob;
+
+import org.springframework.cloud.deployer.scheduler.spi.core.ScheduleRequest;
+import org.springframework.util.Assert;
+
+/**
+ * Configures a {@link Container} that will be launched by a {@link CronJob}.
+ *
+ * @author Chris Schaefer
+ */
+class ContainerCreator {
```

Comment:
most of this stuff already exists in `DefaultContainerFactory` - would be nice to reuse plus be able to take advantage of other functionality in there rather than having to dup it